src/context/Item.tsx

```diff
@@ -0,0 +1,58 @@
+import { createContext, useState } from "react"
+
+export type ListItem = {
+	id?: number, 
+	name: string, 
+	price: number, 
+	link: string, 
+	image: string
+}
+
+export type ItemContextType = {
+	itemList: ListItem[],
+	addItem: (item: ListItem) => void
+}
+
+export const ItemContext = createContext<ItemContextType>({} as ItemContextType)
```

Comment:
```suggestion
export const ItemContext = createContext<ItemContextType>({} as ItemContextType)
```

こちら`as ItemContextType `で無理やり空のオブジェクトを当てはめているのですが、
一旦`defaultItemContext`として定義してから当てはめた方が綺麗だと思いました。

`const ItemContext = createContext<ItemContextType>(defaultItemContext)`こんな感じになればOKです！